salt/tests/unit/modules/test_cri.py

```diff
@@ -276,3 +286,34 @@ def test_ready(self, retcode, result):
         mock_cmd = MagicMock(return_value=retcode)
         with patch.dict(cri.__salt__, {"cmd.retcode": mock_cmd}):
             self.assertEqual(cri.ready(), result)
+
+    @utils.parameterized_from_cases(YAML_TESTS_CASES["stop_pod"])
+    def test_stop_pod(self, result, pod_ids_out=None, pod_stop_out=None, raises=False):
+        """
+        Tests the return of `stop_pod` function
+        """
+
+        def cmd_run_mock(cmd):
+            if cmd.startswith("crictl pods"):
+                return utils.cmd_output(**(pod_ids_out or {}))
+            elif cmd.startswith("crictl stopp"):
+                return utils.cmd_output(**(pod_stop_out or {}))
+            raise Exception("Should not happen !!")
+
+        salt_dict = {"cmd.run_all": MagicMock(side_effect=cmd_run_mock)}
+
+        with patch.dict(cri.__salt__, salt_dict):
+            if raises:
+                self.assertRaisesRegex(
+                    CommandExecutionError,
+                    result,
+                    cri.stop_pod,
+                    {"my.label": "ABCD"},
```

Comment:
Since we can't really test the effect of passing labels to the command, can we at least make sure that the `cmd.run_all` mock receives the expected command lines?

Something like this for `crictl pods`:

```python
# after the call was made
self.assertEqual(
    cmd_run_mock.call_args_list[0][0],
    "crictl pods --quiet --selector=my.label=ABCD"
)
```

Could also add something for `crictl stopp` (comparing with the IDs returned by `crictl pods`), but not necessary I guess.

Reply:
Right I can add it